src/components/StatusWidget.tsx

```diff
@@ -46,32 +47,39 @@ export class StatusWidget extends ReactWidget {
         initialArgs={false}
       >
         {(_, needsCredentials) => (
-          <Badge
-            className={badgeClass}
-            variant="dot"
-            invisible={!needsCredentials}
-            data-test-id="git-credential-badge"
-          >
-            <ActionButton
-              className={classes(
-                toolbarButtonClass,
-                this._status !== 'idle'
-                  ? statusAnimatedIconClass
-                  : statusIconClass
-              )}
-              icon={gitIcon}
-              onClick={
-                needsCredentials
-                  ? async () => this._showGitOperationDialog()
-                  : undefined
-              }
-              title={
-                needsCredentials
-                  ? `Git: ${this._trans.__('credentials required')}`
-                  : `Git: ${this._trans.__(this._status)}`
-              }
-            />
-          </Badge>
+          <>
+            <Badge
+              className={badgeClass}
+              variant="dot"
+              invisible={!needsCredentials}
+              data-test-id="git-credential-badge"
+            >
+              <ActionButton
+                className={classes(
+                  toolbarButtonClass,
+                  this._status !== 'idle'
+                    ? statusAnimatedIconClass
+                    : statusIconClass
+                )}
+                icon={gitIcon}
+                onClick={
+                  needsCredentials
+                    ? async () => this._showGitOperationDialog()
+                    : undefined
+                }
+                title={
+                  needsCredentials
+                    ? `Git: ${this._trans.__('credentials required')}`
+                    : `Git: ${this._trans.__(this._status)}`
+                }
+              />
+            </Badge>
+            {this._model.currentBranch && (
+              <span className={currentBranchClass}>
+                {this._model.currentBranch.name}*
+              </span>
+            )}
```

Comment:
You should place this outside the `UseSignal` component as it does not need to be refreshed if the credentials status of the extension changes.

But you should wrap it in another `UseSignal` component to listen to `this._model.headChanged` to react when the git head is changed.